front/src/common/Map/Layers/GeoJSONs.tsx

```diff
@@ -386,38 +394,54 @@ const GeoJSONs: FC<{
     [hiddenColors, layerContext]
   );
 
-  const sources = useMemo(
-    () =>
-      SOURCES_DEFINITION.flatMap((source) =>
-        !layers || layers.has(source.entityType)
-          ? [
-              {
-                id: `${prefix}geo/${source.entityType}`,
-                url: `${MAP_URL}/layer/${source.entityType}/mvt/geo/?infra=${infraID}`,
-                layers: source
-                  .getLayers({ ...hiddenLayerContext, sourceTable: source.entityType }, prefix)
-                  .map((layer) => adaptFilter(layer, (hidden || []).concat(selection || []), [])),
-              },
-              {
-                id: `${selectedPrefix}geo/${source.entityType}`,
-                url: `${MAP_URL}/layer/${source.entityType}/mvt/geo/?infra=${infraID}`,
-                layers: source
-                  .getLayers({ ...layerContext, sourceTable: source.entityType }, selectedPrefix)
-                  .map((layer) => adaptFilter(layer, hidden || [], selection || [])),
-              },
-            ]
-          : []
-      ),
-    [hidden, hiddenLayerContext, layerContext, layers, infraID, prefix, selectedPrefix, selection]
-  );
+  const sources = useMemo(() => {
+    const res = SOURCES_DEFINITION.flatMap((source) =>
+      !layers || layers.has(source.entityType)
+        ? [
+            {
+              id: `${prefix}geo/${source.entityType}`,
+              url: `${MAP_URL}/layer/${source.entityType}/mvt/geo/?infra=${infraID}`,
+              layers: source
+                .getLayers({ ...hiddenLayerContext, sourceTable: source.entityType }, prefix)
+                .map((layer) => adaptFilter(layer, (hidden || []).concat(selection || []), [])),
+            },
+            {
+              id: `${selectedPrefix}geo/${source.entityType}`,
+              url: `${MAP_URL}/layer/${source.entityType}/mvt/geo/?infra=${infraID}`,
+              layers: source
+                .getLayers({ ...layerContext, sourceTable: source.entityType }, selectedPrefix)
+                .map((layer) => adaptFilter(layer, hidden || [], selection || [])),
+            },
+          ]
+        : []
+    );
+
+    return renderAll
+      ? res.map((source) => ({
+          ...source,
+          layers: source.layers.map((layer) => omit(layer, 'minzoom') as typeof layer),
+        }))
+      : res;
```

Comment:
This test to keep or remove the minzoom should be in the function adaptFilter (maybe we should rename it) + could you add a small comment to precise why this test allows to render all the layers ? 
ex: remove the minzoom props so the layer always renders
```suggestion
    return layers
```